src/main/java/com/linkedin/kmf/services/MultiClusterTopicManagementService.java

```diff
@@ -364,6 +364,20 @@ void maybeAddPartitions(int minPartitionNum) throws ExecutionException, Interrup
       return newPartitionAssignments;
     }
 
+    private static BrokerMetadata randomBroker(Set<BrokerMetadata> brokers) {
```

Comment:
This function is called in a loop and it's `O(n)`. As the list of `brokers` does not change in `newPartitionAssignments`, accepting a `List` argument instead of a `Set` will be faster (`O(1)`)

Reply:
will address your concerns in a sep. PR.

Reply:
https://github.com/linkedin/kafka-monitor/pull/264